entities/card.rb

```diff
@@ -0,0 +1,18 @@
+module Entities
+  class Card
+    attr_accessor :type, :number, :balance, :tax
+
+    def initialize(type:, balance:, tax: 0)
+      @type = type
+      @number = generate_card_number
+      @balance = balance
+      @tax = tax
+    end
+
+    private
+
+    def generate_card_number
+      16.times.map { rand(10) }.join
```

Comment:
Array.new(16) { rand(10) } has a higher performance. Because this one initializes array cells during creating them, and your one firstly creates array with 16 cells and then traverses them and sets new values (rand(10))